File: SDKs/amazonaws.com/cloudfront/2020-05-31/java/lib/src/main/java/openapisdk/models/shared/LambdaFunctionAssociationList.java
package openapisdk.models.shared;



/**
 * LambdaFunctionAssociationList
 * A complex type that contains a Lambda@Edge function association.
**/
public class LambdaFunctionAssociationList {
    public EventTypeEnum eventType;
    public LambdaFunctionAssociationList withEventType(EventTypeEnum eventType) {
        this.eventType = eventType;
        return this;
    }
    public Boolean includeBody;
    public LambdaFunctionAssociationList withIncludeBody(Boolean includeBody) {
        this.includeBody = includeBody;
        return this;
    }
    public String lambdaFunctionARN;
    public LambdaFunctionAssociationList withLambdaFunctionArn(String lambdaFunctionARN) {
        this.lambdaFunctionARN = lambdaFunctionARN;
        return this;
    }
}

File: SDKs/amazonaws.com/cloudfront/2018-06-18/python/src/sdk/models/shared/lambdafunctionassociationlist.py
import dataclasses
from typing import Optional
from enum import Enum
from ..shared import eventtype_enum as shared_eventtype_enum


@dataclasses.dataclass
class LambdaFunctionAssociationList:
    r"""LambdaFunctionAssociationList
    A complex type that contains a Lambda function association.
    """
    
    event_type: shared_eventtype_enum.EventTypeEnum = dataclasses.field()
    lambda_function_arn: str = dataclasses.field()
    include_body: Optional[bool] = dataclasses.field(default=None)
    
